infra/templates/az-isolated-service-single-region/commons.tf

```diff
@@ -6,12 +6,12 @@ data "azurerm_subscription" "current" {}
 
 locals {
   prefix        = "${lower(var.name)}-${lower(terraform.workspace)}"
-  admin_rg_name = "${local.prefix}-admin-rg"                  // name of resource group used for admin resources
-  app_rg_name   = "${local.prefix}-app-rg"                    // name of app resource group
-  sp_name       = "${local.prefix}-sp"                        // name of service plan
-  ai_name       = "${local.prefix}-ai"                        // name of app insights
-  kv_name       = format("%.20s-kv", local.prefix)            // name of key vault
-  acr_name      = replace("${local.prefix}azcr", "/\\W/", "") // name of acr
+  admin_rg_name = "${local.prefix}-admin-rg" // name of resource group used for admin resources
+  app_rg_name   = "${local.prefix}-app-rg"   // name of app resource group
+  sp_name       = "${local.prefix}-sp"       // name of service plan
+  ai_name       = "${local.prefix}-ai"       // name of app insights
+  kv_name       = format("%s%s-kv", format("%.10s", lower(var.name)), format("%.10s", lower(terraform.workspace)))
+  acr_name      = replace(format("%s%sacr", format("%.10s", lower(var.name)), format("%.10s", lower(terraform.workspace))), "/\\W/", "")
   ase_sub_id    = var.ase_subscription_id == "" ? data.azurerm_subscription.current.subscription_id : var.ase_subscription_id
   app_sub_id    = var.app_dev_subscription_id == "" ? data.azurerm_subscription.current.subscription_id : var.app_dev_subscription_id
 
```

Comment:
Changes for this file look great!  Love seeing how replacing special characters in terraform is done.